conf.go

```diff
@@ -248,7 +248,7 @@ func parse(args []string, namespace string, cfgStruct interface{}) error {
 		}
 
 		// If the field is marked 'required', check if no value was provided.
-		if field.Options.Required && !foundOverride {
+		if field.Options.Required && !foundOverride && field.Field.IsZero() {
```

Comment:
These are dangerous changes to make. Have you run through all the tests?

Reply:
Yeah I agree, it does behave weirdly now when adding `IsZero`. One idea forward would be to move away from parsers and add a yaml sourcer instead. What do you think?

Reply:
We will need to do that so the code is isolated from the current path. I am not sure the best way to do that and I don't have time to analyze it.